### PR TITLE
Add Quarkus based Lora adapter

### PR DESCRIPTION
**Pull request by dejanb**

### CODE REVIEW

**Comment by dejanb** [comment]
@BobClaerhout I can't add you as a "reviewer", but would appreciate your feedback

**Comment by BobClaerhout** [comment]
LGTM, I don't see any functional changes.

File: adapters/lora-vertx-quarkus/README.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# hono-adapter-http-vertx-quarkus project
```
Comment (sophokles73) [inline adapters/lora-vertx-quarkus/README.md:1]: this doesn't look right

File: adapters/lora-vertx-quarkus/README.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+## Creating Docker images
+
+You can create a docker image by running: `mvn clean install -Pbuild-docker-image`
+
+If you want to create a native image as well, activate the `native` profile as well:
+
+```sh
+mvn clean install -Pbuild-docker-image,native
+```
```
Comment (sophokles73) [inline adapters/lora-vertx-quarkus/README.md:35]: this should actually refer to the `build-native-image` profile instead ...

File: adapters/lora-vertx-quarkus/src/main/java/org/eclipse/hono/adapter/http/quarkus/MetricsFactory.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+@ApplicationScoped
+public class MetricsFactory {
+
+    @Singleton
+    @Produces
+    MicrometerBasedHttpAdapterMetrics metrics(
+            final Vertx vertx,
+            final MeterRegistry registry,
+            final ProtocolAdapterProperties adapterProperties) {
+        registry.config().commonTags(MetricsTags.forProtocolAdapter(Constants.PROTOCOL_ADAPTER_TYPE_HTTP));
```
Comment (sophokles73) [inline adapters/lora-vertx-quarkus/src/main/java/org/eclipse/hono/adapter/http/quarkus/MetricsFactory.java:39]: PROTOCOL_ADAPTER_TYPE_LORA

File: adapters/lora-vertx-quarkus/src/main/java/org/eclipse/hono/adapter/http/quarkus/Application.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.hono.adapter.http.quarkus;
```
Comment (sophokles73) [inline adapters/lora-vertx-quarkus/src/main/java/org/eclipse/hono/adapter/http/quarkus/Application.java:13]: ...lora.quarkus ?

File: adapters/lora-vertx-quarkus/src/main/java/org/eclipse/hono/adapter/http/quarkus/MetricsFactory.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.hono.adapter.http.quarkus;
```
Comment (sophokles73) [inline adapters/lora-vertx-quarkus/src/main/java/org/eclipse/hono/adapter/http/quarkus/MetricsFactory.java:13]: ...lora.quarkus ?

File: adapters/lora-vertx-quarkus/src/main/java/org/eclipse/hono/adapter/http/quarkus/QuarkusLoraProtocolAdapterProperties.java [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.hono.adapter.http.quarkus;
```
Comment (sophokles73) [inline adapters/lora-vertx-quarkus/src/main/java/org/eclipse/hono/adapter/http/quarkus/QuarkusLoraProtocolAdapterProperties.java:13]: ...lora.quarkus ?

File: adapters/lora-vertx-quarkus/src/main/java/org/eclipse/hono/adapter/http/quarkus/QuarkusLoraProtocolAdapterProperties.java [hunk 7]
```diff
@@ -0,0 +1,25 @@
+/**
+ * Copyright (c) 2020 Contributors to the Eclipse Foundation
```
Comment (sophokles73) [inline adapters/lora-vertx-quarkus/src/main/java/org/eclipse/hono/adapter/http/quarkus/QuarkusLoraProtocolAdapterProperties.java:2]: 2021

File: adapters/lora-vertx-quarkus/src/main/java/org/eclipse/hono/adapter/http/quarkus/MetricsFactory.java [hunk 8]
```diff
@@ -0,0 +1,44 @@
+/**
+ * Copyright (c) 2020 Contributors to the Eclipse Foundation
```
Comment (sophokles73) [inline adapters/lora-vertx-quarkus/src/main/java/org/eclipse/hono/adapter/http/quarkus/MetricsFactory.java:2]: 2021

File: bom/pom.xml [hunk 9]
```diff
@@ -47,6 +47,7 @@
     <javassist.version>3.24.1-GA</javassist.version>
     <jaxb.api.version>2.2.12</jaxb.api.version>
     <javax.annotation.api.version>1.3.2</javax.annotation.api.version>
+    <jakarta.enterprise.cdi-api.version>2.0.2</jakarta.enterprise.cdi-api.version>
```
Comment (sophokles73) [inline bom/pom.xml:50]: this should already be managed by Quarkus. We import Quarkus' BOM pom so there should be no need to specify it here again ...

**Comment by sophokles73** [comment]
@dejanb can you take a look at the failed integration test(s)?

**Comment by dejanb** [comment]
> @dejanb can you take a look at the failed integration test(s)?

Works on my machine (c) :)
There are MQTT test failures, so I'm not sure are related to this PR. I reran the jobs

**Review by sophokles73 — APPROVED** [review]
LGTM

**Pull request merged**